ledfx/effects/random.py

```diff
@@ -0,0 +1,92 @@
+import random
+import time
+
+import numpy as np
+import voluptuous as vol
+
+from ledfx.color import parse_color, validate_color
+from ledfx.effects.temporal import TemporalEffect
+
+
+class RandomEffect(TemporalEffect):
+    """
+    Randomly activates sections
+    """
+
+    NAME = "Random"
+    CATEGORY = "Non-Reactive"
+
+    CONFIG_SCHEMA = vol.Schema(
+        {
+            vol.Optional(
+                "hit_color", description="Hit color", default="#FFFFFF"
+            ): validate_color,
+            vol.Optional(
+                "hit_duration",
+                description="Hit duration",
+                default=0.1,
+            ): vol.All(vol.Coerce(float), vol.Range(min=0.1, max=5.0)),
+            vol.Optional(
+                "hit_probability_per_sec",
+                description="Probability of hit per second",
+                default=0.1,
+            ): vol.All(vol.Coerce(float), vol.Range(min=0.01, max=1.0)),
+            vol.Optional(
+                "hit_relative_size",
+                description="Hit size relative to LED strip",
+                default=0.2,
+            ): vol.All(vol.Coerce(float), vol.Range(min=0.1, max=1.0)),
```

Comment:
You might want to consider a minimum size smaller than 0.1, that is 1 tenth of the overall strip length.

You could consider making this slider to a percentage int, range 1 to 100

Its not consistant in the code base, but for example in the scan effect, I do that in the scan effect

https://github.com/LedFx/LedFx/blob/main/ledfx/effects/scan.py#L44

There are likely example of both styles